legacy/firmware/ethereum.c

```diff
@@ -987,12 +987,27 @@ static void ethereum_typed_hash(const uint8_t domain_separator_hash[32],
   keccak_Final(&ctx, hash);
 }
 
+// When primaryType="EIP712Domain", we ignore message_hash completely
+static void ethereum_domain_only_typed_hash(
```

Comment:
instead, add a boolean parameter to `ethereum_typed_hash()`

Reply:
Added as `bool domain_only_hash` in https://github.com/trezor/trezor-firmware/pull/2036/commits/2d1a145b172ce6e20cce496e9ba2ba05e4863f76, although it's untested, so I'll CI test it for me.